tests/integration/dependencies_test.go

```diff
@@ -0,0 +1,264 @@
+	defer goleak.VerifyNone(t)
+
+	// TODO: Check that probes are really removed if not used anymore
+	testCases := []struct {
+		name             string
+		events           []events.ID
+		expectedLogs     []string
+		expectedEvents   []events.ID
+		unexpectedEvents []events.ID
+	}{
+		{
+			name:           "sanity of submit once test event",
+			events:         []events.ID{events.SubmitOnce},
+			expectedEvents: []events.ID{events.SubmitOnce},
+		},
+		{
+			name:   "non existing ksymbol dependency",
+			events: []events.ID{events.MissingKsymbol, events.SubmitOnce},
+			expectedLogs: []string{
+				"Event canceled because of missing kernel symbol dependency",
+				"Remove event from state",
+			},
+			unexpectedEvents: []events.ID{events.MissingKsymbol},
+			expectedEvents:   []events.ID{events.SubmitOnce},
+		},
+		{
+			name:   "non existing probe function",
+			events: []events.ID{events.FailedAttach, events.SubmitOnce},
+			expectedLogs: []string{
+				"Cancelling event and its dependencies because of a missing probe",
+				"Remove event from state",
+			},
+			unexpectedEvents: []events.ID{events.FailedAttach},
+			expectedEvents:   []events.ID{events.SubmitOnce},
+		},
+	}
+
+	logOutChan, restoreLogger := testutils.SetTestLogger(logger.DebugLevel)
+
+	for _, testCaseInst := range testCases {
+		t.Run(
+			testCaseInst.name, func(t *testing.T) {
+				// prepare tracee config
+				testConfig := config.Config{
+					Capabilities: &config.CapabilitiesConfig{
+						BypassCaps: true,
+					},
+				}
+				testConfig.Policies = buildPoliciesFromEvents(testCaseInst.events)
+				policy.Snapshots().Store(testConfig.Policies)
+
+				ctx, cancel := context.WithCancel(context.Background())
+				done := make(chan struct{})
+				logsResultChan := testLogs(t, testCaseInst.expectedLogs, logOutChan, done)
+
+				// start tracee
+				trc, err := startTracee(ctx, t, testConfig, nil, nil)
+				if err != nil {
+					cancel()
+					t.Fatal(err)
+				}
+				t.Logf("  --- started tracee ---")
+
+				stream := trc.SubscribeAll()
+				defer trc.Unsubscribe(stream)
+				eventsResultChan := testEvents(
+					t,
+					testCaseInst.expectedEvents,
+					testCaseInst.unexpectedEvents,
+					stream.ReceiveEvents(),
+					done,
+				)
+
+				err = waitForTraceeStart(trc)
+				if err != nil {
+					cancel()
+					t.Fatal(err)
+				}
+
+				// Give the events a chance to be received and pass the pipeline
+				time.Sleep(3 * time.Second)
+
+				cancel()
+				errStop := waitForTraceeStop(trc)
+				if errStop != nil {
+					t.Log(errStop)
+				} else {
+					t.Logf("  --- stopped tracee ---")
+				}
+				close(done)
+
+				assert.True(t, <-logsResultChan)
+				assert.True(t, <-eventsResultChan)
+			},
+		)
+	}
+	restoreLogger()
+}
+
+// testLogs searches for given logs and test when input channel closes if all
+// logs were received.
+// It also returns a channel with the result of the test - whether all logs were found.
+func testLogs(
```

Comment:
testing logs is a rather flaky kind of test. Are you sure we need it? If so, lets make the logging based on constants, so the test won't break easily if we change the logs for whatever reason.

Also, since we already added the testutils logger, and _this_ is the way to test it, I think the function should be placed alongside. That way if we do want to keep testing logs, we can reuse this technique.